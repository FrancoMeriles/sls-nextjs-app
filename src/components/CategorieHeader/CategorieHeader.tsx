import { Box, Heading, Image } from '@chakra-ui/react'

const CategorieHeader = ({ name, picture }) => {
  return (
    <Box
      display="flex"
      alignItems="center"
      justifyContent="space-between"
      bg="white"
      borderRadius="10px"
      p={5}
      marginBottom={5}
    >
      <Heading as="h1" size="lg" paddingLeft={5}>
        {name}
      </Heading>
      <Image boxSize="170px" objectFit="cover" src={picture} alt={name} />
    </Box>
  )
}

export default CategorieHeader
