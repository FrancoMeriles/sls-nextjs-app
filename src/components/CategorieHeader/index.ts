import CategorieHeader from './CategorieHeader'

export default CategorieHeader
