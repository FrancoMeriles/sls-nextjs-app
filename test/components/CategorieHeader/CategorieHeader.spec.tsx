import { render, screen } from '@testing-library/react'
import CategorieHeader from '@base/components/CategorieHeader'
import renderer from 'react-test-renderer'

const mockCategorieHeaderProps = {
  picture: '__IMAGE_URL__',
  name: '__NAME__',
}

describe('CategorieHeader', () => {
  it('should render and match snapshot with required props', () => {
    const tree = renderer.create(<CategorieHeader {...mockCategorieHeaderProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should renders and display the correct data', async () => {
    render(<CategorieHeader {...mockCategorieHeaderProps} />)
    expect(screen.getByText(mockCategorieHeaderProps.name)).toBeInTheDocument()
  })
})
