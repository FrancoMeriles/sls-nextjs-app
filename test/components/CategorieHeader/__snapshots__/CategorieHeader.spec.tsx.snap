// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategorieHeader should render and match snapshot with required props 1`] = `
<div
  className="css-iciaa1"
>
  <h1
    className="chakra-heading css-1sugwtq"
  >
    __NAME__
  </h1>
  <img
    alt="__NAME__"
    className="chakra-image__placeholder css-8sjrrl"
  />
</div>
`;
